Tests/MeshLib/TestRasterToMesh.cpp

```diff
@@ -0,0 +1,304 @@
+    {
+        _raster.reset(FileIO::AsciiRasterInterface::readRaster(_file_name));
+    }
+
+    ~RasterToMeshTest() { std::remove(_mesh_name.c_str());  }
+
+protected:
+    std::size_t const _n_pix = 542;
+    std::size_t const _n_nodes = 626;
+    double _spacing = 1000;
+    std::string const _file_name;
+    std::string const _mesh_name;
+    std::unique_ptr<GeoLib::Raster> _raster;
+};
+
+TEST_F(RasterToMeshTest, convertRasterToTriMeshElevation)
+{
+    std::unique_ptr<MeshLib::Mesh> mesh(MeshLib::RasterToMesh::convert(
+        *_raster, MeshLib::MeshElemType::TRIANGLE,
+        MeshLib::UseIntensityAs::ELEVATION, "test"));
+    ASSERT_TRUE(mesh != nullptr);
+
+    MeshLib::IO::VtuInterface vtkio(mesh.get(), 0, false);
+    std::string name(BaseLib::BuildInfo::data_path +
+                     "/MeshLib/testraster_selke.vtu");
+    vtkio.writeToFile(name);
+
+    std::cout << mesh->getNodes().size() << ", " << mesh->getNumberOfNodes()
+              << std::endl;
+    ASSERT_EQ(_n_nodes, mesh->getNodes().size());
+    ASSERT_EQ(_n_nodes, mesh->getNumberOfNodes());
+
+    std::vector<std::string> names =
+        mesh->getProperties().getPropertyVectorNames();
+    ASSERT_TRUE(names.empty());
+
+    std::array<unsigned, 7> n_types =
+        MeshLib::MeshInformation::getNumberOfElementTypes(*mesh);
+    ASSERT_EQ(2 * _n_pix, n_types[1]);
+
+    GeoLib::AABB aabb = MeshLib::MeshInformation::getBoundingBox(*mesh);
+    ASSERT_NEAR(aabb.getMinPoint()[2], 0,
+                std::numeric_limits<double>::epsilon());
+    ASSERT_NEAR(aabb.getMaxPoint()[2], 0.07,
+                std::numeric_limits<double>::epsilon());
+}
+
+TEST_F(RasterToMeshTest, convertRasterToQuadMeshElevation)
+{
+    std::unique_ptr<MeshLib::Mesh> mesh(MeshLib::RasterToMesh::convert(
+        *_raster, MeshLib::MeshElemType::QUAD,  MeshLib::UseIntensityAs::ELEVATION, "test"));
+    ASSERT_TRUE(mesh != nullptr);
+
+    ASSERT_EQ(_n_nodes, mesh->getNumberOfBaseNodes());
+
+    std::vector<std::string> names = mesh->getProperties().getPropertyVectorNames();
+    ASSERT_TRUE(names.empty());
+
+    std::array<unsigned, 7> n_types =
+        MeshLib::MeshInformation::getNumberOfElementTypes(*mesh);
+    ASSERT_EQ(_n_pix, n_types[2]);
+
+    GeoLib::AABB aabb = MeshLib::MeshInformation::getBoundingBox(*mesh);
+    ASSERT_NEAR(aabb.getMinPoint()[2], 0,
+                std::numeric_limits<double>::epsilon());
+    ASSERT_NEAR(aabb.getMaxPoint()[2], 0.07,
+                std::numeric_limits<double>::epsilon());
+}
+
+TEST_F(RasterToMeshTest, convertRasterTo3DMeshElevation)
+{
+    std::unique_ptr<MeshLib::Mesh> mesh(MeshLib::RasterToMesh::convert(
+        *_raster, MeshLib::MeshElemType::PRISM,
+        MeshLib::UseIntensityAs::ELEVATION, "test"));
+    ASSERT_TRUE(mesh == nullptr);
+
+    std::unique_ptr<MeshLib::Mesh> mesh2(MeshLib::RasterToMesh::convert(
+        *_raster.get(), MeshLib::MeshElemType::HEXAHEDRON,
```

Comment:
Of course all occurrences of the `*_raster.get()` should be changed to `*_raster`....